sdk/macro/src/lib.rs

```diff
@@ -426,6 +426,7 @@ pub fn derive_clone_zeroed(input: proc_macro::TokenStream) -> proc_macro::TokenS
             let name = &item_struct.ident;
             quote! {
                 impl Clone for #name {
+                    #[allow(clippy::incorrect_clone_impl_on_copy_type)]
```

Comment:
So this lint: https://rust-lang.github.io/rust-clippy/master/index.html#/incorrect_clone_impl_on_copy_type
> If both Clone and Copy are implemented, they must agree. This is done by dereferencing self in Clone’s implementation. Anything else is incorrect.

In this case, we are "manually" implementing `Clone` to make sure all padding bytes are zeroed out.
Based on that description I don't think we'll ever be able to remove the allow, since it's a correct flagging of this scenario.
IIRC rust makes no guarantees for `Copy` in terms of padding bytes bein zero (probably not since it's just a memcpy?) - would defer to @Lichtso since he's the one who added the macro.

In anycase I will add a comment on why this allow is there/necessary.

Reply:
9b08354991

Reply:
We want a custom copy operation. Probably should derive both copy and clone, I guess.

Reply:
It's not possible to derive a custom `Copy` in rust as far as I am aware - it's a special trait that effectively allows types to be `memcpy`'d

Reply:
Re-thinking about this, I'm not sure we even want these types to be `Copy`. It's probably safer to only support `Clone` w/ our zeroed padding stuff.
Otherwise it doesn't really protect us from sending something over the network w/ non-zero padding because we could have just copied our value instead of cloning?

Reply:
We can try unimplementing `Copy` for these types, but that might result in a lot of `.clone()` where implicit deref happened before.

Reply:
👍 I think that sounds reasonable. I'm going to add this clippy-lint allow for now, and just self-assign an issue for experimenting with removing `Copy`.
Without removing `Copy` this allow is necessary to make clippy happy.

Reply:
#32588